src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,144 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as os from 'os';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as mssql from 'vscode-mssql';
+import { defaultBindingResult, defaultSqlBindingTextLines, genericCollectionImport, sqlBindingResult } from '../constants/constants';
+import * as LocalizedConstants from '../constants/localizedConstants';
+import { AzureFunctionsService } from '../services/azureFunctionsService';
+import * as azureFunctionUtils from '../azureFunction/azureFunctionUtils';
+import * as constants from '../constants/constants';
+import * as path from 'path';
+
+export class AzureFunctionProjectService {
+
+	constructor(private azureFunctionsService: AzureFunctionsService) {
+	}
+
+	public async createAzureFunction(connectionString: string, schema: string, table: string): Promise<void> {
+		const azureFunctionApi = await azureFunctionUtils.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			return;
+		}
+		if (!await isAzureFunctionProjectOpen()) {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+			return;
+		}
+
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = getNewFunctionFile();
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: constants.functionNameTitle,
+			value: 'HttpTrigger1'
+		});
+
+		// create C# HttpTrigger
+		await azureFunctionApi.createFunction({
+			language: 'C#',
+			templateId: 'HttpTrigger',
+			functionName: functionName
+		});
+
+		// once steps are addressed we add SQL Bindings Nuget and Connection String to local.settings.json
+		// and then create a new functions file
+		await addNugetReferenceToProjectFile();
```

Comment:
I think we need to revisit how this all works. Currently it's kind of odd that the logic is split up into two places : 

1. The extension handles stuff like adding the nuget reference, adding the connection string config and this whole "refactoring" stuff
2. STS handles actually adding the sql binding

There doesn't seem to be any reason why we would split that logic up. Having STS handle everything seems to make more sense to me - it has a lot more power (such as access to the rosyln parser) to make code modifications and keeping things in one place (especially a common place like STS) means that if anything else wants to use it we don't have to re-implement this logic elsewhere (say if we wanted to add this functionality into ADS for example).

@alanrenmsft @abist Do you know why this was done like this originally?

I'm doing some work that is going to need to use this functionality which I would greatly prefer to have all in STS if possible. Now since we've done the work here already I guess we can get it in like this for the first step, but if there's nothing blocking it then I'd like to see the next step after merging this in be to move the logic over to STS (the logic itself can mostly stay the same so that shouldn't be a huge task).

@chlafreniere Thoughts?

Reply:
I can setup a meeting tomorrow / async discussion on this in order to get this resolved sooner rather than later.